packages/clarity-js/src/data/upload.ts

```diff
@@ -126,8 +126,8 @@ function stringify(encoded: EncodedPayload): string {
 
 function send(payload: string, sequence: number, last: boolean): void {
     // Upload data if a valid URL is defined in the config
-    if (typeof config.upload === Constant.String) {
-        const url = config.upload as string;
+    if (typeof config.upload === Constant.String && config.server) {
```

Comment:
Thsi seems like it wouldn't be backward compatible. Will stop uploading everything on older config.